bootstrapvz/plugins/admin_user/tasks.py

```diff
@@ -42,13 +43,62 @@ def run(cls, info):
+                from bootstrapvz.common.exceptions import TaskError
+                from bootstrapvz.common.tools import sed_i
+                from bootstrapvz.common.tools import log_check_call
+                log = logging.getLogger(__name__)
+
+                if 'password' in info.manifest.plugins['admin_user']:
+                        if 'pubkey' in info.manifest.plugins['admin_user']:
+                                msg = 'The options password and pubkey are mutually exclusive'
+                                raise TaskError(msg)
+                        log_check_call(['chroot', info.root, 'chpasswd'],
+                                       info.manifest.plugins['admin_user']['username'] +
+                                       ':' + info.manifest.plugins['admin_user']['password'])
+                        return
+
+                getcreds_path = os.path.join(info.root, 'etc/init.d/ec2-get-credentials')
+                if 'pubkey' in info.manifest.plugins['admin_user']:
+                        import stat
+                        from shutil import copy
+                        full_path = info.manifest.plugins['admin_user']['pubkey']
+                        if not os.path.exists(full_path):
+                                msg = 'Could not find public key at {full_path}'.format(full_path=full_path)
+                                raise TaskError(msg)
+                        log.debug('Copying public key from {path}'.format(path=full_path))
+
+                        if os.path.exists(getcreds_path):
+                                log.warn('You are using a static public key for the admin account.'
+                                         ' This will conflict with the ec2 public key njection mechanisn.'
+                                         ' The ec2-get-credentials startup script has therefore been disabled.')
+                                log_check_call(['chroot', info.root, 'insserv', '--remove',
+                                                'ec2-get-credentials'])
+                        username = info.manifest.plugins['admin_user']['username']
+
+                        ssh_file = os.path.join('/home/'
+                                                '{username}/.ssh/authorized_keys'.format(username=username))
+                        rel_ssh_file = os.path.realpath(info.root + '/{ssh_file}'.format(ssh_file=ssh_file))
+
+                        ssh_dir = os.path.dirname(ssh_file)
+                        rel_ssh_dir = os.path.realpath(info.root + '/{ssh_dir}'.format(ssh_dir=ssh_dir))
+                        if not os.path.exists(rel_ssh_dir):
+                                log.debug('Creating {ssh_dir} mode 700'.format(ssh_dir=rel_ssh_dir))
+                                os.mkdir(rel_ssh_dir, 0700)
+                        else:
+				log.debug('setting {ssh_dir} mode 700'.format(ssh_dir=rel_ssh_dir))
+                                os.chmod(rel_ssh_dir, 0700)
```

Comment:
python
Python 2.7.11 (default, Dec  9 2015, 00:29:25) 
[GCC 5.3.1 20151205] on linux2
os.chmod('/tmp/junk', 0700)
>>> 
% ls -als ll /tmp/junk
0 -rwx------ 1 srivasta srivasta 0 Jan  8 22:29 /tmp/junk

So it seems to work, at least on Debian and Ubuntu